File: backend-python/app/auth/claims.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Claims:
    """user_id解決(UserResolver)にはsub/issのみ必要だが、外部公開API(RequireExternalClientAuth)は
    azp(authorized party)クレームでClient Credentials Grantのクライアントを識別する必要があるため保持する
    """

    sub: str
    iss: str
    azp: str = ""


class VerifyException(Exception):
    pass

File: backend-elixir/lib/backend_elixir/auth/claims.ex
defmodule BackendElixir.Auth.Claims do
  @moduledoc """
  user_id解決(UserResolver)に必要な最小限のクレームに、外部公開API向けの`azp`を追加している。
  `azp`(authorized party)はClient Credentials Grantで発行されたトークンのクライアントIDを表し、
  外部公開API(External.Handler)がこの値をEXTERNAL_API_CLIENT_IDと比較する(内部REST/gRPCの
  user_id解決には使わない)
  """
  defstruct [:sub, :iss, :azp]
end
